index.js

```diff
@@ -1,59 +1,90 @@
-const express = require('express');
+const express = require("express");
+const server = express();
 const cors = require("cors");
 const bcrypt = require("bcryptjs");
 const db = require("./db/dbConfig.js");
+const session = require("express-session");
+const config = require("./config/middleware");
 
-const server = express();
+const KnexSessionStore = require("connect-session-knex")(session);
 
 server.use(express.json());
 server.use(cors());
+// server.use();
+config(server); 
 
+const sessionConfig = {
+  name: "banana", // default is connect.sid
+  secret: "nobody tosses a dwarf!",
+  cookie: {
+    maxAge: 1 * 24 * 60 * 60 * 1000,
+    secure: false // only set cookies over https. Server will not send back a cookie over http.
+  }, // 1 day in milliseconds
+  httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+  resave: false,
+  saveUninitialized: false,
+  store: new KnexSessionStore({
+    tablename: "sessions",
+    sidfieldname: "sid", //what is this line of code?
+    knex: db,
+    createtable: true,
+    clearInterval: 1000 * 60 * 60
+  })
+};
 
-server.get('/', (req, res) => {
-    res.send("hello world");
-})
-
+server.use(session(sessionConfig));
 
-server.post("/api/register", (req, res) => {
-    const creds = req.body;
+auth = (req, res, next) => {
```

Comment:
It's a _real good_ idea to preface your variable declarations with `let` or `const`. Otherwise, JS will treat this variable declaration in a weird way, leading to mystery errors and headaches.